File: python/7 kyu/353.py
# https://www.codewars.com/kata/57f75cc397d62fc93d000059/train/python

# Passed

def calc(x):
    character_codes = [str(ord(character)) for character in x]
    total1 = int("".join(character_codes))
    total2 = int(str(total1).replace("7", "1"))
    
    get_digits_sum = lambda num: sum([int(digit) for digit in str(num)])
    
    result = get_digits_sum(total1) - get_digits_sum(total2)
    return result

output = calc('abcdef')
print(output)

File: nodejs/6 kyu/353.js
// https://www.codewars.com/kata/590938089ff3d186cb00004c/train/javascript

// Passed

function suffixSums(a) {
  let result = [];
  let currentVal = 0;
  for (let idx = a.length - 1; idx >= 0; idx--) {
    currentVal += a[idx];
    result.push(currentVal);
  }

  result = result.reverse();
  return result;
}

const output = suffixSums([1, 123, 23]);
console.log(output);